authors/apps/articles/test/test_bookmark.py

```diff
@@ -0,0 +1,145 @@
+import json
+from .base import BaseTestArticles
+from rest_framework.views import status
+from ..models import Article
+from ...authentication.models import User
+from ....apps.profiles.models import Profile
+
+
+class TestBookmark(BaseTestArticles):
+
+    def test_bookmark_successful(self):
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + self.login_user())
+        response = self.client.post(
+            '/api/articles/', data=self.article, format='json')
+        slug = response.data['slug']
+        res = self.client.post(
+            '/api/articles/{}/bookmark/'.format(slug),
+            format='json'
+        )
+        self.assertEqual(res.status_code, 201)
+        self.assertGreater(len(res.data), 1)
+
+    def test_article_was_already_bookmarked(self):
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + self.login_user())
+        response = self.client.post(
+            '/api/articles/', data=self.article, format='json')
+        slug = response.data['slug']
+        self.client.post(
+            '/api/articles/{}/bookmark/'.format(slug),
+            format='json'
+        )
+        res = self.client.post(
+            '/api/articles/{}/bookmark/'.format(slug),
+            format='json'
+        )
+        self.assertEqual(res.status_code, 301)
+        self.assertEqual('article was bookmarked', res.data['message'])
+
+    def test_article_does_not_exist(self):
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + self.login_user())
+        self.client.post(
+            '/api/articles/', data=self.article, format='json')
+        slug = 'Fake slug'
+        res = self.client.post(
+            '/api/articles/{}/bookmark/'.format(slug),
+            format='json'
+        )
+        self.assertEqual(res.status_code, 404)
+        self.assertEqual("Article does not exist", res.data['error'])
+
+    def test_get_bookmarked_articles(self):
```

Comment:
Looks like this tests getting a single bookmark. Should be `test_get_bookmarked_article`

Reply:
thanks. I have added the right endpoint for getting all the articles